### PR TITLE
Add metadata api wrapper methods

### PR DESCRIPTION
**Pull request by czarpino**
What this PR introduces:

* New DataRecords for Metadata `MetadataSetRecord` & `MetadataValueRecord`
* New CloudApi methods `getAvailableMetadataSets` & `getMetadataValues`
* Unit tests
* New directory structure for PSR4 adoption

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Looked through the PR, looks mighty fine - the PSR4 autoload dir structure is a great addition.

**Comment by jefersonralmeida** [comment]
Hey @czarpino ! I completely understood your logic. Really good implementation. But there is a potential problem...
We have the entire api on a single file for a reason..., some customers just download this file, and requires it on their system (that was the original instruction on how to use this wrapper, before composer). So I don't think we can create separated files (not without talking with Anis or Madhan).

A possible solution would be converting this giant file on a loader, and split the current content into multiple files, updating the version of the library (to 1.1 for example, or 2.0), requiring the customers to download the entire content but including on their codebase just this loader. It's important to keep this approach because not everyone uses composer...

**Review by jefersonralmeida — DISMISSED** [review]
Like I said, having multiple files need to be discussed before been implemented. Marking as "needs work" just to prevent the merging before we have a deeper discussion about this...

**Review by tomfor — DISMISSED** [review]
I fully agree with @jefersonralmeida on this one. We should discuss it separately. In my opinion, at some point, we should simply reorganize the API into smaller, logical classes, do some refactoring, potentially with the cleaner return types, add type hinting, annotations and release it as a new API wrapper, i.e. version 2.0.

**Comment by czarpino** [comment]
Thanks @jefersonralmeida, I understand. That is quite an important detail! I guess transitioning into the usual 1 class per file will require more consideration and planning.

**Comment by czarpino** [comment]
> A possible solution would be converting this giant file on a loader, and split the current content into multiple files, updating the version of the library (to 1.1 for example, or 2.0), requiring the customers to download the entire content but including on their codebase just this loader

I think this is a good approach if users are willing to make the slight adjustment ("slight" being possibly relative). Although if the benefits of the API wrapper are "that" big, users just may put up with the discomfort of migrating.

**Comment by czarpino** [comment]
> In my opinion, at some point, we should simply reorganize the API into smaller, logical classes, do some refactoring, potentially with the cleaner return types, add type hinting, annotations and release it as a new API wrapper, i.e. version 2.0.

Makes sense. A major version bump along with major changes. I think this is where we ultimately want to go to. 🙂

File: fccloudapi.php [hunk 1]
```diff
@@ -171,6 +171,449 @@ public function getObjectName()
     }
 }
 
+/**
+ * Base MetadataRecord class
+ * @package codelathe\fccloudapi
+ */
+abstract class AbstractMetadataRecord extends DataRecord
```
Comment (tomfor) [inline fccloudapi.php:178]: Why is that named the AbstractMetadataRecord? This is in fact the attribute value converter.

**Review by tomfor — DISMISSED** [review]
Everything looks good

File: fccloudapi.php (hunk 1 above)
Comment (czarpino) [reply fccloudapi.php:178]: You may be right. I investigated making the change but it seems the current way makes testing a bit easier so I'd like to keep it like this for now.

**Review by jefersonralmeida — DISMISSED** [review]

**Review by kaynetik — DISMISSED** [review]

**Review by kaynetik — APPROVED** [review]

**Review by tomfor — APPROVED** [review]

**Pull request merged**